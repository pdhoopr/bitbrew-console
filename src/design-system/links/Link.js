import styled from "styled-components";
import LinkImpl from "./LinkImpl";
import linkStyles from "./linkStyles";

export default styled(LinkImpl)`
  ${linkStyles};
  border-bottom: var(--border-transparent);
  color: ${({ green }) => green && "var(--color-green)"};
  transition: border-bottom-color var(--effect-duration),
    color var(--effect-duration);

  &:hover,
  &:focus {
    border-bottom-color: currentColor;
    color: var(--color-green);
  }
`;
