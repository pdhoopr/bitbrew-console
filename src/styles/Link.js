import styled from 'styled-components';

export default styled.a`
  border-bottom: 1px solid transparent;
  color: var(--color-green);
  cursor: pointer;
  text-decoration: none;
  transition: border-bottom-color var(--duration-short),
    color var(--duration-short);

  &:hover {
    border-bottom-color: currentColor;
  }
`;
