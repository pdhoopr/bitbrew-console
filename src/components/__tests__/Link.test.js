import { shallow } from 'enzyme';
import React from 'react';
import Link from '../Link';

test('renders DOM for a link', () => {
  const dom = shallow(<Link href="https://bitbrew.com" />);
  expect(dom).toMatchSnapshot();
});
