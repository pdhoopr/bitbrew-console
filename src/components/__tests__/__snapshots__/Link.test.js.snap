// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders DOM for a link 1`] = `
.c0 {
  border-bottom: 1px solid transparent;
  color: inherit;
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: border-bottom-color var(--duration-short),color var(--duration-short);
  transition: border-bottom-color var(--duration-short),color var(--duration-short);
}

.c0:hover {
  border-bottom-color: currentColor;
  color: var(--color-green);
}

<a
  className="c0"
  href="https://bitbrew.com"
/>
`;
